README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

## Features

A flutter package to give you a customize appbar with search .

## Getting started

This project is a starting point for a Dart package, a library module containing code that can be shared easily across multiple Flutter or Dart projects.

For help getting started with Flutter, view our online documentation, which offers tutorials, samples, guidance on mobile development, and a full API reference.

## Usage

A simple usage example:

```dart
      IconButton(
        icon: Icon(Icons.clear),
        onPressed: () {
          if (query.isEmpty) {
            close(context, null);
          } else {
            query = "";
          }
        },
      )
```

CHANGELOG.md
## 0.0.1

AppBar Search


test/ic_searchbar_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:ic_searchbar/ic_searchbar.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


lib/src/api.dart
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'dart:async';

import 'employees_list.dart';

class FetchUser {
  var data = [];
  List<Employees> results = [];
  String fetchurl = "https://jsonplaceholder.typicode.com/users";
  
  Future<List<Employees>> getEmployees({String? query}) async {
    var url = Uri.parse(fetchurl);
    var response = await http.get(url);
    try {
      if (response.statusCode == 200) {
        data = json.decode(response.body);
        results = data.map((e) => Employees.fromJson(e)).toList();
        if (query !=null){
          results = results.where((element) => element.name!.toLowerCase().contains(query.toLowerCase())).toList();
        }
      } else {
        print('api error');
      }
    } on Exception catch (e) {
      print('No results found');
    }
    return results;
  }
}


lib/src/employees_list.dart
class Employees {
  int? id;
  String? name;
  String? username;
  String? email;
  Address? address;
  String? phone;
  String? website;
  Company? company;

  Employees(
      {this.id,
      this.name,
      this.username,
      this.email,
      this.address,
      this.phone,
      this.website,
      this.company});

  Employees.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    username = json['username'];
    email = json['email'];
    address =
        json['address'] != null ? new Address.fromJson(json['address']) : null;
    phone = json['phone'];
    website = json['website'];
    company =
        json['company'] != null ? new Company.fromJson(json['company']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['username'] = this.username;
    data['email'] = this.email;
    if (this.address != null) {
      data['address'] = this.address?.toJson();
    }
    data['phone'] = this.phone;
    data['website'] = this.website;
    if (this.company != null) {
      data['company'] = this.company?.toJson();
    }
    return data;
  }

  static void where(Null Function() param0) {}
}

class Address {
  String? street;
  String? suite;
  String? city;
  String? zipcode;
  Geo? geo;

  Address({this.street, this.suite, this.city, this.zipcode, this.geo});

  Address.fromJson(Map<String, dynamic> json) {
    street = json['street'];
    suite = json['suite'];
    city = json['city'];
    zipcode = json['zipcode'];
    geo = json['geo'] != null ? new Geo.fromJson(json['geo']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['street'] = this.street;
    data['suite'] = this.suite;
    data['city'] = this.city;
    data['zipcode'] = this.zipcode;
    if (this.geo != null) {
      data['geo'] = this.geo?.toJson();
    }
    return data;
  }
}

class Geo {
  String? lat;
  String? lng;

  Geo({this.lat, this.lng});

  Geo.fromJson(Map<String, dynamic> json) {
    lat = json['lat'];
    lng = json['lng'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['lat'] = this.lat;
    data['lng'] = this.lng;
    return data;
  }
}

class Company {
  String? name;
  String? catchPhrase;
  String? bs;

  Company({this.name, this.catchPhrase, this.bs});

  Company.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    catchPhrase = json['catchPhrase'];
    bs = json['bs'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = this.name;
    data['catchPhrase'] = this.catchPhrase;
    data['bs'] = this.bs;
    return data;
  }
}


lib/src/individual_list.dart
import 'package:flutter/material.dart';
import 'api.dart';
import 'employees_list.dart';

class IndividualList extends StatefulWidget {
  const IndividualList({super.key});

    @override
  State<IndividualList> createState() => _IndividualList();
}

class _IndividualList extends State<IndividualList> {
  FetchUser _fetchUser = FetchUser();
  
  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        appBar: AppBar(
          title: Text('back'),
        ),
        body: Container(
          child: FutureBuilder<List<Employees>>(
              future: _fetchUser.getEmployees(),
              builder: (context, snapshot) {
                var data = snapshot.data;

                return ListView.builder(
                  itemCount: data?.length,
                  itemBuilder: (context, index) {
                    return Card(
                      child: ListTile(
                        title: Row(
                          children: [
                            Container(
                             height: 45,
                              width: 45,
                              decoration: BoxDecoration(
                                color: Color.fromARGB(255, 71, 166, 239),
                                borderRadius: BorderRadius.circular(50),
                              ),
                              child: Center(
                                child: Text(
                                  '${data?[index].id}',
                                  style: TextStyle(
                                      color: Colors.white,
                                      fontWeight: FontWeight.bold),
                                ),
                              ),
                            ),
                            SizedBox(width: 10),
                            Column(
                              crossAxisAlignment: CrossAxisAlignment.start,
                              children: [
                                Text(
                                  '${data?[index].name}',
                                  style: TextStyle(fontWeight: FontWeight.w600),
                                ),
                                Text('${data?[index].email}'),
                              ],
                            )
                          ],
                        ),
                      ),
                    );
                  },
                );
                
              }),
        ),
      ),
    );
  }
}

lib/src/search.dart
import 'package:flutter/material.dart';
import 'api.dart';
import 'employees_list.dart';
import 'individual_list.dart';

class SearchUser extends SearchDelegate {
  @override
  List<Widget>? buildActions(BuildContext context) {
    return [
      IconButton(
        icon: Icon(Icons.clear),
        onPressed: () {
          if (query.isEmpty) {
            close(context, null);
          } else {
            query = "";
          }
        },
      )
    ];
    throw UnimplementedError();
  }

  @override
  Widget? buildLeading(BuildContext context) {
    return IconButton(
      icon: Icon(Icons.arrow_back_ios),
      onPressed: () {
        // Navigator.pop(context);
        close(context, null);
      },
    );
  }

  final FetchUser _fetchUser = FetchUser();

  @override
  Widget buildResults(BuildContext context) {
    return Container(
      child: FutureBuilder<List<Employees>>(
          future: _fetchUser.getEmployees(query: query),
          builder: (context, snapshot) {
            var data = snapshot.data;

            return ListView.builder(
              itemCount: data?.length,
              itemBuilder: (context, index) {
                return Card(
                  child: ListTile(
                    title: Row(
                      children: [
                        Container(
                       height: 45,
                              width: 45,
                          decoration: BoxDecoration(
                            color: Color.fromARGB(255, 71, 166, 239),
                            borderRadius: BorderRadius.circular(50),
                          ),
                          child: Center(
                            child: Text(
                              '${data?[index].id}',
                              style: TextStyle(
                                  color: Colors.white,
                                  fontWeight: FontWeight.bold),
                            ),
                          ),
                        ),
                        SizedBox(width: 10),
                        Column(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Text(
                              '${data?[index].name}',
                              style: TextStyle(fontWeight: FontWeight.w600),
                            ),
                            Text('${data?[index].email}'),
                          ],
                        )
                      ],
                    ),
                  ),
                );
              },
            );
          }),
    );
  }

  @override
  Widget buildSuggestions(BuildContext context) {
    // return Center(
    //   child: Text('Search Users'),
    // );

    return Container(
      child: FutureBuilder<List<Employees>>(
          future: _fetchUser.getEmployees(query: query),
          builder: (context, snapshot) {
            switch (snapshot.connectionState) {
              case ConnectionState.waiting:
                return Center(child: CircularProgressIndicator());
              default:
                if (query.isEmpty) return buildNoSuggestions();
                var data = snapshot.data;

                if (snapshot.hasError || snapshot.data!.isEmpty) {
                  return buildNoSuggestions();
                } else {
                  return ListView.builder(
                    itemCount: data?.length,
                    itemBuilder: (context, index) {
                      return Card(
                        child: ListTile(
                          onTap: () => Navigator.push(
                            context,
                            MaterialPageRoute(
                                builder: (context) => IndividualList()),
                          ),
                          title: Row(
                          children: [
                            Container(
                              child: Center(
                                child: 
                                Icon(Icons.search)
                              ),
                            ),
                            SizedBox(width: 10),
                            Column(
                              crossAxisAlignment: CrossAxisAlignment.start,
                              children: [
                                Text(
                                  '${data?[index].name}',
                                  style: TextStyle(fontWeight: FontWeight.w600,),
                                ),
                                Text('${data?[index].email}', style: TextStyle(color: Color.fromARGB(255, 175, 172, 172),)),
                              ],
                            )
                          ],
                        ),
                        ),
                      );
                    },
                  );
                }
            }
          }),
    );
  }
}

Widget buildNoSuggestions() => const Center(
      child: Text(
        'No Data found!',
        style: TextStyle(fontSize: 20),
      ),
    );

